### PR TITLE
Add snackbar new features

### LINKED ISSUE #518: Add snackbar related to new features
## Description

Add snackbar in forms and places that might be relevant to give user feedback.

## Possible Implementation

use enqueueSnackbar from notistack's useSnackbar in functional compoennt

### PR DESCRIPTION
**Pull request by AnnieTran13**
Closes #518 

Snackbar added for:
- Campaign create/edit success
- Event register/leave success

### CODE REVIEW

File: resources/js/Components/Form/CampaignForm.tsx [hunk 1]
```diff
@@ -59,7 +61,26 @@ const CampaignForm = ({
                   adventure: editData?.adventure || undefined,
               }
 
-    const {data, setData, errors, put, post} = useForm<CampaignFormDataType>(CAMPAIGN_INITIAL_VALUE)
+    const {data, setData, errors, clearErrors, put, post, wasSuccessful} =
```
Comment (willyyhuang) [inline resources/js/Components/Form/CampaignForm.tsx:64]: can you also add `processing` as well for loading button here?

Comment (AnnieTran13) [reply resources/js/Components/Form/CampaignForm.tsx:64]: on it!

File: resources/js/Components/Modal/EventRegistrationModal.tsx [hunk 2]
```diff
@@ -65,16 +68,25 @@ const EventRegistrationModal = ({
         wasSuccessful,
         processing,
     } = useForm<FormDataType>(SESSION_REGISTRATION_FORM_INITIAL_VALUE)
+    const {enqueueSnackbar} = useSnackbar()
 
     useEffect(() => {
         if (wasSuccessful) {
             clearErrors()
-            onClose()
+            if (onClose) {
```
Comment (willyyhuang) [inline resources/js/Components/Modal/EventRegistrationModal.tsx:76]: why do we need this btw? `onClose` is always required

Comment (AnnieTran13) [reply resources/js/Components/Modal/EventRegistrationModal.tsx:76]: I'm not sure why it's there but I'll make the change!

**Review by willyyhuang — APPROVED** [review]

**Pull request merged**